### PR TITLE
Option --flatKeys

### PR DESCRIPTION
**Pull request by atufkas**
Hi Keyang,

#### TL;DR
*Here's a pull request implementing `--flatKeys` allowing flat JSON key-value associations without interpretation of dots and square brackets in header fields as discussed in #51*

#### Details...

I'd like to kindly ask you to review this pull request and integrate it if you feel it's clean and worth it. It is related to issue No. #51 where a "flat interpretation" of header fields is demanded. Same situation here in my case: While the JSON empowered input filter feature is great, I have to traverse some "original" and untouched CSV data to just a simple one-dimensional JSON object at first hand. Dots may occur and shouldn't be interpreted as a nesting delimiter.

I first started with adding two new options `--noNesting` and `--noArrays` but then realized that it makes things complicated (and not too much sense) when disabling one feature while still allowing the other. So my approach now is the new option `--flatKeys` which just disables both: Dots and square brackets in header fields of the default *json* parser. 

From what I understand looking at your code + docs, the default *json* parser should be capable of handling most input gracefully, so that there's normally no need for implementing custom callbacks or even parser. That's the reason I implemented this as an option instead of a new built-in parser.

I added some test and additional docs.

Thanks in Advance & Happy Coding,
Matthias

### CODE REVIEW

**Comment by Keyang** [comment]
Brilliant. 
Thanks very much for the contribution.
I will take a look.

Regards,
Keyang

**Comment by Keyang** [comment]
Hi,
I read through the code.
It is very good and does the work -- turn off nesting JSON construction and use Flat key.
I actually have another solution which is adding a parser "*flat*" which will just pass the header as it is to construct a flat JSON object. E.g.

```csv
*flat*user.name, *flat*user.age, region.city,region.country
Joe, 40, London, UK
```

will be converted to:

```json
[{
"user.name":"Joe",
"user.age":40,
"region":{
    "city":"London",
    "country":"UK"
}
}]
```

so developers can set specific column as "flat" while keep other columns "nested". Or, they can mark all headers as "flat". 

What is your option on this?

Regards,
Keyang

**Comment by atufkas** [comment]
Hi Keyang,

thanks for your quick review! 

Yes, it's true that your approach is probably a better fit in the philosophy of dynamically attaching "parser modules" (built-in or custom) that may cover specific needs. I hesitated a while exactly for that reason before committing my code changes following my approach. 

So, why an option `--flatKeys`? I had an "working out-of-the-box" approach in mind where people can easily "flatly" test and import their arbitrary data without the need of initially defining the structure in a `headers` string. Think of data like:

```
Id, Product No., Title, Price
1, 4711, XBox One, $349.00
2. 5517, Playstation 4, $299.00
```

So a simple `csv2json --flatKeys=true product.csv` would do the job as a quick start.

Of course it's nice to have the option for selectively defining the handling on a per-column basis.  You should decide about it. Maybe it would be nice to have your idea implemented plus an option `--flatKeys` which can always be overridden by custom column markers and appropriate parsers?

Up to you! I'd like to code and contribute anyways if you made a decision :)

Regards,
Matthias

**Comment by Keyang** [comment]
Great. Let's do both ways and let the devs to pick whatever they would like to use.
I will merge the changes when I get the time and publish a new version along with the new default parser "flat".

Regards,
Keyang

**Comment by atufkas** [comment]
:+1:

**Comment by Keyang** [comment]
I have merged your changes.
Feel free to add yourself as a contributor in package.json if you want.

Thanks,
Keyang
